### PR TITLE
 [#196][feat] add where condition to manifests

### PR DESCRIPTION
**Pull request by icepuma**
- use rhai scripts to use condition
  - if the rhai script resolves to true -> run manifest
  - if the rhai script fails or resolves to false -> don't run manifest
- context variables can be used

## I'm submitting a

- [ ] bug fix
- [x] feature

## What is the motivation / use case for changing the behavior?

If a manifest runs or not depends on the `dry run` option. Manifest execution can be fine tuned by adding `where` conditions.

### CODE REVIEW

**Comment by rawkode** [comment]
This is a different implementation that we use for actions and I'm wondering if we need them to be comparable?

So far, for actions we support a single string and that string can be evaluated with multiples only within that string, such as:

`variables.name == "David" && variables.location != "The Moon"`

We should make manifests do this, or update actions to be a list too?

Or maybe it's not a big deal. I'd like to hear thoughts

**Comment by icepuma** [comment]
Yeah, @theowenyoung was asking for `or` operations. Therefore I will make it a one-liner. Then it's possible to let the user decide the logic behind `where`.

**Comment by icepuma** [comment]
The good thing is, that yaml supports multiline strings and therefore more or less the thing I tried to model with the multiple where conditions

**Comment by icepuma** [comment]
I changed from multiple conditions to one. Feels better and more consistent.

File: tests/basic_usage.rs [hunk 1]
```diff
@@ -32,6 +32,19 @@ actions:
     to: mydircopy
 "#,
                 ),
+                f(
+                    "where_condition.yaml",
+                    r#"
+where:
+  - non.existing.variable == true
```
Comment (rawkode) [inline tests/basic_usage.rs:39]: This seems like the old revision with multiple

**Review by rawkode — CHANGES_REQUESTED** [review]

File: tests/basic_usage.rs (hunk 1 above)
Comment (icepuma) [reply tests/basic_usage.rs:39]: Argh. Thanks. That's what reviews are for!

**Review by rawkode — APPROVED** [review]
If the build passes, merge at your own convenience. Great addition.

**Pull request merged**